### PR TITLE
added new tests to cover openstackmap scripts

### PR DESCRIPTION
**Pull request by geonexus**
### REVIEWERS
@flopezag @chemaper @jframos 

### DESCRIPTION
- Adding more unittests to cover osclients script.

### CODE REVIEW

File: tests/units/test_openstackmap.py [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+import os
+from requests import Response
+
+
+class MySessionMock(MagicMock):
+    # Mock of a keystone Session
+
+    def get_endpoint(self, auth, **kwargs):
+
+        if kwargs.get("service_type") == "volume":
```
Comment (flopezag) [inline tests/units/test_openstackmap.py]: else?

File: tests/units/test_openstackmap.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+                                   {u'url': u'http://172.0.0.1:4731/v3/', u'interface': u'administator',
+                                    u'region': u'Spain2', u'id': u'00000000000000000000000000000001'}],
+                                    u'type': u'identity', u'id': u'00000000000000000000000000000045'}
+
+        d = defaultdict(list)
+        d['catalog'].append(service2)
+
+        return d
+
+    def request(self, url, method, json=None, original_ip=None,
```
Comment (flopezag) [inline tests/units/test_openstackmap.py]: Do we need all of those parameters?

File: tests/units/test_openstackmap.py (hunk 1 above)
Comment (geonexus) [inline tests/units/test_openstackmap.py]: fixed

File: tests/units/test_openstackmap.py (hunk 2 above)
Comment (geonexus) [inline tests/units/test_openstackmap.py]: Fixed

File: tests/units/test_openstackmap.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+                 '"links": [{"href": "http://83.32.33.1:8774/00000000000000000000000000000001/flavors/4", ' \
+                 '"rel": "bookmark"}]}, "id": "00000000-0001-1c9a-v77t-1234567abcdf",' \
+                 ' "security_groups": [{"name": "default"}], "OS-SRV-USG:terminated_at": null, ' \
+                 '"OS-EXT-AZ:availability_zone": "nova", "user_id": "user", "name": "Main", ' \
+                 '"created": "2015-11-23T08:08:49Z", "tenant_id": "00000000000000000000000000000001",' \
+                 ' "OS-DCF:diskConfig": "MANUAL", "os-extended-volumes:volumes_attached": [], "accessIPv4": "",' \
+                 ' "accessIPv6": "", "progress": 0, "OS-EXT-STS:power_state": 1, "config_drive": "",' \
+                 ' "metadata": {"region": "Spain2"}}]}'
+
+            resp.status_code = 200
```
Comment (flopezag) [inline tests/units/test_openstackmap.py]: MAGIC NUMBER

File: tests/units/test_openstackmap.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+        d['catalog'].append(service2)
+
+        return d
+
+    def request(self, url, method, **kwargs):
+
+        resp = Response()
+
+        if url == '/servers/detail?all_tenants=1':
+            text = '{"servers": [{"status": "ACTIVE", "updated": "2015-11-23T08:08:58Z", ' \
```
Comment (flopezag) [inline tests/units/test_openstackmap.py]: from my point of view, I usually take out in a file all the text content in order to clarify the text and put these files inside the resource directory. @jframos what do you think?

File: tests/units/test_openstackmap.py [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+
+    def test_implement_openstackmap_without_region_name(self):
+        """test_implement_openstackmap_without_region_name check that we could not build an empty map without providing
+        a region Name."""
+
+        del os.environ['OS_REGION_NAME']
+        try:
+            myopenstackmap = OpenStackMap(auth_url=self.OS_AUTH_URL, auto_load=False)
+        except Exception as ex:
+            self.assertRaises(ex)
```
Comment (jframos) [inline tests/units/test_openstackmap.py]: To check that an exception is raised when callable is called, I think following way is better than using _try/except_ blocks as part of unit tests:

       with self.assertRaises(Exception):
            OpenStackMap(auth_url=self.OS_AUTH_URL, auto_load=False)

File: tests/units/test_openstackmap.py (hunk 4 above)
Comment (geonexus) [inline tests/units/test_openstackmap.py]: i think that it is not the same. It is a json response expected from different requests

Comment (jframos) [inline tests/units/test_openstackmap.py]: :+1:

Especially whether they are large requests/responses.
Perhaps you might put in `tests/unit/resources` different _.json_ templates with this data.

Comment (geonexus) [inline tests/units/test_openstackmap.py]: Refactoring done!

File: tests/units/test_openstackmap.py (hunk 3 above)
Comment (geonexus) [inline tests/units/test_openstackmap.py]: Fixed

File: tests/units/tests_constants.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+UNIT_TEST_RESOURCES_FOLDER = 'tests/units/resources/'
+LIST_SERVERS_RESPONSE_FILE = 'list_servers_response.json'
+LIST_VOLUMES_RESPONSE_FILE = 'list_volumes_response.json'
+LIST_SNAPSHOTS_RESPONSE_FILE = 'list_snapshots_response.json'
+LIST_ROLES_RESPONSE_FILE = 'list_roles_response.json'
+LIST_BACKUPS_RESPONSE_FILE = 'list_backups_response.json'
+LIST_USERS_RESPONSE_FILE = 'list_users_response.json'
+LIST_PROJECTS_RESPONSE_FILE = 'list_projects_response.json'
+LIST_ROLE_ASSIGNMENTS_RESPONSE_FILE = 'list_role_assignments_response.json'
```
Comment (flopezag) [inline tests/units/tests_constants.py]: add new line

File: tests/units/test_openstackmap.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+         resources providing a keystone admin endpoint environment variable."""
+
+        environ.setdefault('KEYSTONE_ADMIN_ENDPOINT', self.OS_AUTH_URL)
+
+        mock_exists.return_value = False
+        mock_os.return_value = True
+        openstackmap = OpenStackMap(auto_load=False)
+        self.assertTrue(mock_os.called)
+        self.assertIsNotNone(openstackmap)
+        del os.environ['KEYSTONE_ADMIN_ENDPOINT']
```
Comment (jframos) [inline tests/units/test_openstackmap.py]: Should be this operation in a 'tearDown' method?

Comment (geonexus) [inline tests/units/test_openstackmap.py]: I don't think so, because there are different test cases and that sentence is not used in all methods.

**Comment by jframos** [comment]
LGTM

**Comment by flopezag** [comment]
LGTM

**Pull request merged**